domains/pallets/evm-tracker/src/check_nonce.rs

```diff
@@ -73,6 +72,9 @@ where
     type Pre = Pre;
 
     fn weight(&self, _: &T::RuntimeCall) -> sp_weights::Weight {
+        // Currently we use the benchmarked weight from `frame_system::CheckNonce`, which does
+        // slightly less work. This extension does one more `max()` and two extra comparisons.
+        // TODO: benchmark evm_tracker::check_nonce() directly and use that weight here.
```

Comment:
TBH, I dont think it actually makes any difference but if we want to sure we can

Reply:
I don't think it matters much, 3 operations is a trivial amount (a few hundred picoseconds if they're not optimised out). But it's useful to document why using the upstream weight is justified.